src/commands/server/update.ts

```diff
@@ -75,108 +77,78 @@ export default class Update extends Command {
     return path.join(__dirname, '../../../templates')
   }
 
-  async checkIfInstallerSupportUpdating(flags: any) {
+  async run() {
+    const { flags } = this.parse(Update)
+    const ctx = await initializeContext(flags)
+    await this.setDomainFlag(flags)
     if (!flags.installer) {
       await this.setDefaultInstaller(flags)
       cli.info(`› Installer type is set to: '${flags.installer}'`)
     }
 
-    if (flags.installer === 'operator' || flags.installer === 'olm') {
-      // operator already supports updating
-      return
-    }
-
-    if (flags.installer === 'olm' && flags.platform === 'minishift') {
-      this.error(`🛑 The specified installer ${flags.installer} does not support Minishift`)
-    }
-  }
-
-  async run() {
-    const { flags } = this.parse(Update)
-    const ctx = initializeContext()
-    const listrOptions: Listr.ListrOptions = { renderer: (flags['listr-renderer'] as any), collapse: false } as Listr.ListrOptions
-    ctx.listrOptions = listrOptions
     ctx.CRPatch = readCRPatchFile(flags, this)
 
-    const cheTasks = new CheTasks(flags)
     const kubeHelper = new KubeHelper(flags)
-    const platformTasks = new PlatformTasks()
     const installerTasks = new InstallerTasks()
-    const apiTasks = new ApiTasks()
-
-    // Platform Checks
-    const platformCheckTasks = new Listr(platformTasks.preflightCheckTasks(flags, this), listrOptions)
-    platformCheckTasks.add(CommonPlatformTasks.oAuthProvidersExists(flags))
 
-    await this.checkIfInstallerSupportUpdating(flags)
-
-    // Checks if Eclipse Che is already deployed
-    let preInstallTasks = new Listr(undefined, listrOptions)
-    preInstallTasks.add(apiTasks.testApiTasks(flags, this))
-    preInstallTasks.add({
-      title: '👀  Looking for an already existing Eclipse Che instance',
-      task: () => new Listr(cheTasks.checkIfCheIsInstalledTasks(flags, this))
-    })
-
-    const preUpdateTasks = new Listr(installerTasks.preUpdateTasks(flags, this), listrOptions)
-
-    const updateTasks = new Listr(undefined, listrOptions)
+    // pre update tasks
+    const apiTasks = new ApiTasks()
+    const preUpdateTasks = new Listr([], ctx.listrOptions)
+    preUpdateTasks.add(apiTasks.testApiTasks(flags, this))
+    preUpdateTasks.add(CommonPlatformTasks.oAuthProvidersExists(flags))
+    preUpdateTasks.add(installerTasks.preUpdateTasks(flags, this))
+
+    // update tasks
+    const updateTasks = new Listr([], ctx.listrOptions)
+    updateTasks.add(apiTasks.testApiTasks(flags, this))
```

Comment:
We had already put `apiTasks.testApiTasks(flags, this)` to the `preUpdateTasks`. Do we really need to put it to the `apiTasks.testApiTasks` one more time?

Reply:
Fixed